vaadin-select-flow/src/main/java/com/vaadin/flow/component/select/Select.java

```diff
@@ -452,14 +493,35 @@ public void setDataProvider(DataProvider<T, ?> dataProvider) {
         return dataProvider.get();
     }
 
+    @Override
+    public SelectDataView<T> setDataSource(DataProvider<T, ?> dataProvider) {
+        this.setDataProvider(dataProvider);
+        return getDataView();
+    }
 
     @Override
-    public SelectListDataView<T> setDataProvider(
+    public SelectListDataView<T> setDataSource(
             ListDataProvider<T> dataProvider) {
-        this.setDataProvider((DataProvider<T, ?>) dataProvider);
+        this.setDataProvider(dataProvider);
         return getListDataView();
     }
 
+    /**
+     * Getter for getting a generic SelectDataView. This should be used
+     * only when
+     * neither {@link #getListDataView()} nor #getLazyDataView are applicable
```

Comment:
>     * neither {@link #getListDataView()} nor #getLazyDataView are applicable

This javadocs is broken but also `getLazyDataView` doesn't apply for `Select` I think ever.

Reply:
Changed